app/FAIRshakeWeb/view.py

```diff
@@ -57,47 +57,48 @@ def register():
 
 @app.route('/project/<int:project>/resources', methods=['GET'])
 def resources(repository: RepositoryAPI, assessment: AssessmentAPI, score: ScoreAPI, project):
-  resources=repository.get(tag=project)
-  current_user_assessments = []
-  assessment_count = {}
-  aggregate_scores = {}
-  for resource in resources: #resource-id=1-7
-    try:
-      assessment.get(object=resource.id, user=current_user['id']) ### redesign, if there is an assessment for this object for this user
-      current_user_assessments.append(resource.id) # just array with resource ids
-    except:
-      pass
-    assessment_count[resource.id] = len(assessment.get(object=resource.id)) # all assessments from all users for all rubrics
-    aggregate_scores[resource.id] = score.get(id=resource.id,kind='text/html') # scores is an array of html with tooltip script
-  current_user_assessments=['1'] ###
+  resources=repository.get(tags=[project])
+  current_user_assessed_resources = [
+    resource.id for resource in resources
+    if assessment.get(object=resource.id, user=current_user['id']) != []
+  ]
+  assessment_count = {
+    resource.id:len(assessment.get(object=resource.id))
+    for resource in resources
+  }
+  aggregate_scores = {
+    resource.id:score.get(id=resource.id,kind='text/html')
+    for resource in resources
+  }
   return render_template('project_resources.html',
-    project=repository.get(id=project,limit=1), ###
+    project=repository.get(id=project,limit=1)[0],
```

Comment:
You can use the `first` function in utils (or `first_and_only` to assert that there is only one)--the function has the benefit of working on generators.

Reply:
I will address this